### PR TITLE
MOTOR-769  MOTOR-770 Fix test_fetch_next_exception and test_exhaust

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
Both of these test failures were caused because the tests muck around with internal pymongo state around cursor closing and cursor del. The way we handle those changed subtly in PyMongo 3.12 so the tests started failing. There's no pymongo bug here.

**Comment by ShaneHarvey** [comment]
The Windows failures are expected.

File: test/asyncio_tests/test_asyncio_cursor.py [hunk 1]
```diff
@@ -467,16 +467,24 @@ async def test_exhaust(self):
         sock = one(socks)
         cur = client[self.db.name].test.find(
             cursor_type=CursorType.EXHAUST).batch_size(1)
-        has_next = await cur.fetch_next
-        self.assertTrue(has_next)
+        await cur.fetch_next
+        self.assertTrue(cur.next_object())
```
Comment (prashantmital) [inline test/asyncio_tests/test_asyncio_cursor.py:471]: Can't we write this test in a way that it doesn't use these deprecated methods? We really should be testing this with `async for`

File: test/asyncio_tests/test_asyncio_grid_file.py [hunk 2]
```diff
@@ -122,9 +122,6 @@ async def test_readchunk(self):
 
     @asyncio_test
     async def test_gridout_open_exc_info(self):
-        if sys.version_info < (3, ):
```
Comment (prashantmital) [inline test/asyncio_tests/test_asyncio_grid_file.py:125]: Nice 👍🏽

**Review by prashantmital — COMMENTED** [review]
Do we have a ticket already to eliminate duplication of test code between asyncio and tornado test modules? It is entirely unnecessary now that we only support Python 3 and Tornado has deprecated its own coroutine syntax in favor of asyncio so the APIs for concrete API classes are now identical...

File: test/asyncio_tests/test_asyncio_cursor.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/asyncio_tests/test_asyncio_cursor.py:471]: Yes we could but I think that's out of scope for this ticket.

Comment (prashantmital) [reply test/asyncio_tests/test_asyncio_cursor.py:471]: It is? I was just suggesting not adding any new test code that uses deprecated methods, but OK.

**Review by prashantmital — APPROVED** [review]

**Comment by ShaneHarvey** [comment]
> Do we have a ticket already to eliminate duplication of test code between asyncio and tornado test modules? It is entirely unnecessary now that we only support Python 3 and Tornado has deprecated its own coroutine syntax in favor of asyncio so the APIs for concrete API classes are now identical...

I opened https://jira.mongodb.org/browse/MOTOR-772

File: test/asyncio_tests/test_asyncio_cursor.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/asyncio_tests/test_asyncio_cursor.py:471]: This test is already using fetch_next/next_object. Using async-for would be out of place and rewriting the whole test to use async-for is a much larger change, something that should happen in MOTOR-605.

**Pull request merged**